frontend/app/professor/simulation-builder/page.tsx

```
@@ -695,7 +764,258 @@ export default function ScenarioBuilder() {
+        }
+        
+        // Skip if this figure has a role that suggests they're the main protagonist
+        const protagonistRoles = ['protagonist', 'main character', 'lead', 'principal', 'central figure'];
+        if (protagonistRoles.some(role => figureRole.includes(role))) {
+          return false;
+        }
+        
+        return true;
+      });
+      
+      const newPersonas = filteredFigures.map((figure: any, index: number) => {
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)